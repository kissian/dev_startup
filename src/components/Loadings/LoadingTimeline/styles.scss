@mixin nth-children($points...) {
  @each $point in $points {
      &:nth-child(#{$point}) {
          @content;
      }
  }
}


.LoadingUser {
  padding: 15px;
  display: flex;
  flex-direction: column;
  height: 214px;
  border: 1px solid #eee;
  border-radius: 0;
  width: 100%;
  margin-bottom: 30px;
  .loadingTitle {
    height: 10px;
    margin: 10px 0;
    width: 70%;
    animation: pulse 1s infinite ease-in-out;
  }
  .row {
    padding: 7px 15px;
    display: flex;
    flex-direction: row;
    align-items: center;
    .loadingCircle {
      border-radius: 100%;
      width: 20px; height: 20px;
      margin: 0 10px;
      animation: pulse 1s infinite ease-in-out;
      
    }
    .loadingLine {
      height: 10px;
      margin: 0 10px;
      animation: pulse 1s infinite ease-in-out;
      @include nth-children(1,3,5,7) {
        width: 20%;
      }
      @include nth-children(2,4,6,8) {
        width: 40%;
      }
    }
  }
}

@keyframes pulse {
  0% {
      background-color: rgba(165, 165, 165, 0.1)
  }
  50% {
      background-color: rgba(165, 165, 165, 0.3)
  }
  100% {
      background-color: rgba(165, 165, 165, 0.1)
  }
}