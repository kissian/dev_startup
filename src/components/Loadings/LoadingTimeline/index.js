import React from 'react';
import './styles.scss';

const LoadingTimeline = () => (
    <div className="flex-column timelineLoding">
        <div className="flex-row-center">
            <div style={{ marginRight: 12 }} className="loadingCircle" />
            <div style={{ flex: '1 1 auto' }}>
                <div className="flex-row-center">
                    <div className="loadingLine" style={{ flex: '1 1 0' }} />
                </div>
                <div className="flex-row-center">
                    <div style={{ flex: '1 1 0' }}>
                        <div className="loadingLine" style={{ flex: '1 1 0' }} />
                    </div>
                </div>
            </div>
        </div>
        <div style={{ marginTop: 6, flex: '1 1 0' }}>
            <div className="loadingLine" style={{ flex: '1 1 0' }} />
            <div className="loadingLine" style={{ flex: '1 1 0' }} />
            <div className="loadingLine" style={{ flex: '1 1 0' }} />
            <div className="loadingBox" style={{ flex: '1 1 0' }} />
        </div>
    </div>
);

export default LoadingTimeline;
