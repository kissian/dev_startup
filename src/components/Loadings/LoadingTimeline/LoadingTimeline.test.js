import React from 'react';
import { shallow } from 'enzyme';
import LoadingTimeline from './index';

describe('LoadingTimeline component', () => {
    let component = null;
    it('renders component', () => {
        component = shallow(<LoadingTimeline />);
    });
    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});
