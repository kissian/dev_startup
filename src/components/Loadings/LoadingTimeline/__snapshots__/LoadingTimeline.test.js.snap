// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingTimeline component matches snapshot 1`] = `
<div
  className="flex-column timelineLoding"
>
  <div
    className="flex-row-center"
  >
    <div
      className="loadingCircle"
      style={
        Object {
          "marginRight": 12,
        }
      }
    />
    <div
      style={
        Object {
          "flex": "1 1 auto",
        }
      }
    >
      <div
        className="flex-row-center"
      >
        <div
          className="loadingLine"
          style={
            Object {
              "flex": "1 1 0",
            }
          }
        />
      </div>
      <div
        className="flex-row-center"
      >
        <div
          style={
            Object {
              "flex": "1 1 0",
            }
          }
        >
          <div
            className="loadingLine"
            style={
              Object {
                "flex": "1 1 0",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "1 1 0",
        "marginTop": 6,
      }
    }
  >
    <div
      className="loadingLine"
      style={
        Object {
          "flex": "1 1 0",
        }
      }
    />
    <div
      className="loadingLine"
      style={
        Object {
          "flex": "1 1 0",
        }
      }
    />
    <div
      className="loadingLine"
      style={
        Object {
          "flex": "1 1 0",
        }
      }
    />
    <div
      className="loadingBox"
      style={
        Object {
          "flex": "1 1 0",
        }
      }
    />
  </div>
</div>
`;
